src/Scrabble/Search.hs

```diff
@@ -54,6 +56,22 @@ containsAll :: String -> Search
 containsAll s1 s2 = fst $ foldl' f (True, ups s2) (ups s1) where
   f (b,s) c = if elem c s then (b, delete c s) else (False,s)
 
+-- | Like the above, but if it encounters a character in in s1 not in
+--   s2, it deletes a blank and tries again before returning false
+containsAllWithBlanks :: String -> Search
```

Comment:
I think having `Search` and `ScrabbleSearch` is clear enough, `SearchBase` sounds a bit redundant to me. I like the idea though, I could see these functions being distinct enough to warrant their own module, even thought implementation-wise they're really close to `containsAll`